internal/kafka/handler.go

```diff
@@ -140,6 +141,16 @@ func validateRequestID(requestIDLength int, requestID string) bool {
 	return true
 }
 
+func attemptPayloadInsertion(db *gorm.DB, payloadStatus *models.PayloadStatuses) error {
+	result := queries.InsertPayloadStatus(db, payloadStatus)
+
+	if result.Error != nil {
+		return result.Error
+	}
+
+	return nil
```

Comment:
I think you could simplify this by just returning `result.Error`.  At that point, maybe it doesn't make sense to have this function.